device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -244,9 +244,20 @@ public void open() throws TransportException
                     closeConnectionWithException("Timed out waiting for worker links to open", true);
                 }
             }
+            catch (TransportException e)
+            {
+                // clean up network resources and thread scheduler before exiting this layer. Subsequent open attempts
+                // will create a new reactor and a new executor service
+                this.reactor.free();
+                this.executorServicesCleanup();
+                throw e;
+            }
             catch (InterruptedException e)
             {
-                this.close();
```

Comment:
Close() has some logic that isn't necessary to run when the connection wasn't opened successfully, so instead of calling close, I just moved the necessary logic up to these catch blocks